flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/FunctionGenerator.scala

```diff
@@ -753,6 +753,54 @@ object FunctionGenerator {
     Seq(FLOAT, INTEGER),
     BuiltInMethods.TRUNCATE_FLOAT)
 
+  addSqlFunctionMethod(
+    IS_JSON_VALUE,
+    Seq(CHAR),
```

Comment:
I think `VARCHAR` will be safer than `CHAR`  here.

Reply:
Thanks for your suggestion. Is it ditto for the rest of `CHAR` in these declarations?

Reply:
Yes.

Reply:
`FunctionGenerator` is not perfect now, we may need to declare different combinations of types. We will improve that in the future. But I think `VARCHAR` can suite both for CHAR and VARCHAR here. So let's reduce the combinations here.

Reply:
Unfortunately after I change `CHAR` to `VARCHAR` the test cases `'[]'` and so on failed with mismatch. Thus I manually add `CHAR` *and* `VARCHAR` in `FunctionGenerator`. I agree that we can do follow-ups to ease the overhead of combinations.